### PR TITLE
Dont restrict scroll when layerTarget present

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Does not restrict page scroll when a layerTarget is present. A previously merged PR monitored for scroll event and that was working for the tested implementation. However, if a `Grommet` component is provided with `height="auto"` (to properly calculate the height of the entire page), suddenly the page still would not scroll. In order to overcome this, we need to adjust the `restrictScroll` prop, that is used internally for Layer in modal cases, to only be true if a Layer does not have a target.

#### Where should the reviewer start?
src/js/components/Layer/LayerContainer.js

#### What testing has been done on this PR?
Tested locally with a long page with Layer with a target on a Box contained in the page. Grommet component of the example was given `style={{height: 'auto'}}`. 

#### How should this be manually tested?
Same as above.

#### Any background context you want to provide?
Modal layer + target was restricting scroll on pages with height="auto".

#### What are the relevant issues?
https://github.com/grommet/hpe-design-system/issues/1115

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/Layer/LayerContainer.js [hunk 1]
```diff
@@ -163,7 +163,7 @@ const LayerContainer = forwardRef(
       content = (
         <FocusedContainer
           hidden={position === 'hidden'}
-          restrictScroll
+          restrictScroll={!layerTarget ? true : undefined}
```
Comment (ericsoderberghp) [inline src/js/components/Layer/LayerContainer.js:166]: Can you add some comments here, just so we don't lose the "why"?

Comment (taysea) [reply src/js/components/Layer/LayerContainer.js:169]: I've added a comment for future reference.

**Review by ericsoderberghp — APPROVED** [review]

**Comment by ShimiSun** [comment]
Can we add any tests to these varied bug fixes of scrolling when using Layer? Could also be relevant to #4755/

**Comment by taysea** [comment]
> Can we add any tests to these varied bug fixes of scrolling when using Layer? Could also be relevant to #4755/

I am not sure that we can since capturing scroll behavior isn't reliable in jest. it would be more suited to test in an e2e environment where the tests are running in a browser environment.

**Comment by ShimiSun** [comment]
@taysea undestood. Can we create a test on the DS site that ensures the correct behavior?

**Comment by taysea** [comment]
> @taysea understood. Can we create a test on the DS site that ensures the correct behavior?

Yes I will make sure to add one in once I implement the work on the site that leverages this work.

**Pull request merged**